src/test/java/com/pnu/dev/shpaltaif/PublicAccountServiceIntegrationTest.java

```diff
@@ -134,64 +129,108 @@ public void createAndThenReadAndThenUpdateAndThenDeleteSuccessFlow() {
-        PublicAccount updatedPublicAccount = publicAccountService.findById(publicAccount1.getId());
-        assertEquals(updatedPublicAccount.getPseudonym(), publicAccountDtoWithPseudonym.getPseudonym().trim());
-        assertEquals(updatedPublicAccount.isPseudonymUsed(), publicAccountDtoWithPseudonym.isPseudonymUsed());
+        ServiceException thrown = assertThrows(ServiceException.class,
+                () -> publicAccountService.update(publicAccountDtoWithBlankSignature, publicAccountFromDb1.getId()));
+        assertEquals(thrown.getMessage(), "Щоб використовувати псевдонім, введіть його коректно");
 
-        PublicAccount publicAccount2 = createUserWriter("username2").getPublicAccount();
-        PublicAccountDto publicAccountDtoWithExistsPseudonym = PublicAccountDto.builder()
-                .name(publicAccount2.getName())
-                .surname(publicAccount2.getSurname())
+        //Set correct pseudonym to publicAccount1
+        String pseudonym = "pseudonym";
+        PublicAccountDto publicAccountDto = PublicAccountDto.builder()
+                .name(publicAccountFromDb1.getName())
+                .surname(publicAccountFromDb1.getSurname())
                 .pseudonymUsed(true)
-                .pseudonym("Pseudonym")
+                .pseudonym(pseudonym)
                 .build();
+        publicAccountService.update(publicAccountDto, publicAccountFromDb1.getId());
 
+        //Set booked pseudonym to publicAccount2
+        publicAccount = createPublicAccount("name", "surname");
+        PublicAccount publicAccountFromDb2 = publicAccountService.findById(publicAccount.getId());
         thrown = assertThrows(ServiceException.class,
-                () -> publicAccountService.update(publicAccountDtoWithExistsPseudonym, publicAccount2.getId()));
+                () -> publicAccountService.update(publicAccountDto, publicAccountFromDb2.getId()));
         assertEquals(thrown.getMessage(), "Псевдонім зайнятий");
     }
 
-    private User createUserWriter(String username) {
+    private PublicAccount createPublicAccount(String name, String surname) {
 
-        int usersNumberBeforeCreate = userRepository.findAll().size();
+        User user = createUserWriter();
 
-        CreateUserDto createUserDto = CreateUserDto
-                .builder()
-                .username(username)
-                .password("password")
-                .repeatedPassword("password")
-                .name("name")
-                .surname("surname")
+        PublicAccountDto publicAccountDto = PublicAccountDto.builder()
+                .name(name)
+                .surname(surname)
                 .build();
 
-        userService.create(createUserDto);
-        List<User> allUsers = userRepository.findAll();
-        assertEquals(usersNumberBeforeCreate + 1, allUsers.size());
+        return publicAccountService.create(publicAccountDto, user);
+    }
+
+    private User createUserWriter() {
```

Comment:
no need in this method, just move this code to the method above.
I'd even suggest moving this code to the tests methods itself, along with account creation (actually, it's not a duplicate, it's test preparation, and the simplier it will look like, the better it will be).